python/cudf/groupby/groupby.py

```diff
@@ -298,15 +297,15 @@ def apply_multiindex_or_single_index(self, result):
                 else:
                     new_by.append(by)
             self._by = new_by
-            multi_index = MultiIndex(source_data=result[self._by])
+            mi = MultiIndex(names=self._by, source_data=result[self._by])
```

Comment:
@kkraus14 I think this assumed the MultiIndex infers names from the `source_data` columns. The logic about where names come from was a bit tangled, so I updated this call-site to be more explicit.